internal/cli/cmd/cluster/buildproxy.go

```diff
@@ -181,8 +182,9 @@ func (bp *buildProxy) Cleanup() error {
 }
 
 func (bp *buildProxy) Serve(ctx context.Context) error {
-	if err := serveProxy(ctx, bp.listener, func(ctx context.Context) (net.Conn, error) {
-		return bp.instance.NewConn(ctx)
+	sink := tasks.SinkFrom(ctx)
```

Comment:
Can we put this behind a flag so we can test the new implementation in staging for a bit?